app/src/main/java/org/wikipedia/analytics/eventplatform/EventPlatformClient.java

```diff
@@ -239,14 +237,27 @@ static void schedule(Event event) {
         }
 
         /**
-         * If sending is enabled, attempt to send the provided event.
-         *
-         * @param event event
+         * If sending is enabled, attempt to send the provided events.
+         * Also batch the events ordered by their streams, as the QUEUE
+         * can contain events of different streams
+         * @param events list of events
          */
-        static void send(Event event) {
-            postEvent(getStreamConfig(event.getStream()), event);
+        static void send(List<Event> events) {
```

Comment:
A function called `send()` should do one thing: send.  If there are conditions that might cause it not to send, it should be called `maybeSend()`, or `sendIfEnabled()`.

For maximum clarity, the check for `ENABLED` should probably be the responsibility of the caller of `send()`, instead of within it.

Reply:
Correct. The enable check might be redundant here, since the caller is checking. So removing the check and keeping the name.